Testing/Tests/LinkedList_Test.cpp

```diff
@@ -22,6 +22,14 @@ TEST_GROUP(LinkedList)
 
    void teardown()
    {
+      LinkedListNode_t *temp;
+      LinkedListNode_t *head = instance.head;
+      while(head != NULL)
+      {
+         temp = head;
+         head = head->next;
+         free(temp);
+      }
```

Comment:
If you choose to use `malloc`, it would be better to provide a `LinkedList_Destroy` function that handles this logic in the library, not on the client side.